c++/src/kj/compat/http.h

```diff
@@ -52,10 +52,7 @@ namespace kj {
   MACRO(PURGE) \
   MACRO(OPTIONS) \
   MACRO(TRACE) \
-  /* standard methods */ \
-  /* */ \
-  /* (CONNECT is intentionally omitted since it should be handled specially in HttpServer) */ \
```

Comment:
It's kind of precarious to add CONNECT here because it means we have to be very careful that the application cannot use it. For example, `HttpClient::request()` really needs to throw if given `CONNECT` as the method, otherwise request smuggling vulnerabilities may open up. It's also dangerous if an HTTP server tries to process a `CONNECT` request as if it were a generic method.

I do see that this creates a problem for `HttpHeaders::tryParseRequest()`... we would want that to support parsing CONNECT requests, too. Maybe the solution is to extend the `RequestOrProtocolError` `OneOf` to include `ConnectRequest` as a completely separate type? That way no one can accidentally handle `CONNECT` the same as everything else.

Reply:
Makes sense. Ok, I'll take another pass.

Reply:
Adding a new option to `RequestOrProtocolError` is going to have a trickle down effect on the rest of the public API so I want to make sure it's what we want. Specifically, on the `HttpInputStream` API, the `readRequest()` method current returns a `kj::Promise<Request>`. We'll need that to learn the difference between a regular request and a connect request. There are two options:

1. We modify the `HttpInputStream::Request::method` field to be `kj::OneOf<HttpMethod, HttpConnectMethod>` (`HttpConnectMethod` is new)
2. We modify the return value of `HttpInputStream::readRequest()` to be `kj::Promise<kj::OneOf<Request, ConnectRequest>>`

Do you have a preference?

Reply:
Yes, #3 is the right answer. This is a public interface meant as a helper for certain oddball applications that aren't actually HTTP servers but want to read HTTP-shaped requests. Such applications are getting an error thrown today when the request is CONNECT, so continuing to throw an error at them is fine.

If some app actually wants to use this interface to receive CONNECT requests, then we can add a new method for them to be able to do that. But for now we should not bother.

`HttpServer` itself does not use the public `HttpInputStream` interface, so we can add CONNECT support without changing that interface.